metadata_layer.py

```diff
@@ -1909,9 +1911,11 @@ def _key(classification):
                 continue
 
             if replace.mirror:
-                # We need to mirror this resource. If it's an image, a
-                # thumbnail may be provided as a side effect.
-                self.mirror_link(edition, data_source, link, link_obj, replace)
+                for mirror_type in replace.mirror.keys():
+                    if replace.mirror[mirror_type]:
+                        # We need to mirror this resource. If it's an image, a
+                        # thumbnail may be provided as a side effect.
+                        self.mirror_link(edition, data_source, link, link_obj, replace, mirror_type)
```

Comment:
Again, I think every mirror is going to mirror every link. And this looks like duplicate code (though it was duplicate code before you changed it). Maybe the right thing to do is to put all the code in `mirror_link`, and make `mirror_link` figure out whether the link should actually be mirrored, and if so, which is the right mirror to use.